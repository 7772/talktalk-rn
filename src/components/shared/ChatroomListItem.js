// @flow
import React, { Component } from 'react';
import {
  StyleSheet,
  Image,
  Text,
  View,
  TouchableOpacity,
} from 'react-native';
import moment from 'moment';
import Icon5 from 'react-native-vector-icons/FontAwesome5';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import { Chatroom } from '../../utils/Types';
import { colors } from '../../utils/Styles';

type Styles = {
  container: ViewStyle,
  wrapper: ViewStyle,
  status: ViewStyle,
  imgWrapper: ViewStyle,
  img: ImageStyle,
  viewContent: ViewStyle,
  viewTop: ViewStyle,
  txtDisplayName: TextStyle,
  viewCount: ViewStyle,
  txtCount: TextStyle,
  viewBottom: ViewStyle,
  txtMsg: TextStyle,
  txtDate: TextStyle,
};

const styles: Styles = StyleSheet.create({
  container: {
    width: '100%',
  },
  wrapper: {
    backgroundColor: 'white',
    height: 80,
    borderBottomWidth: 1,
    borderColor: 'rgb(247,248,251)',

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  status: {
    position: 'absolute',
    width: 12,
    height: 12,
    borderRadius: 6,
    backgroundColor: colors.greenishCyan,
    right: 0,
    bottom: 0,
    borderWidth: 2,
    borderColor: 'white',
  },
  imgWrapper: {
    marginHorizontal: 20,
  },
  img: {
    width: 40,
    height: 40,
  },
  viewContent: {
    flexDirection: 'column',
    flex: 1,
    paddingRight: 20,
  },
  viewTop: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  txtDisplayName: {
    fontSize: 14,
    color: colors.dusk,
  },
  viewCount: {
    width: 16,
    height: 16,
    borderRadius: 8,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgb(80,227,194)',
  },
  txtCount: {
    fontSize: 10,
    color: 'white',
  },
  viewBottom: {
    marginTop: 8,

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  txtMsg: {
    fontSize: 12,
    color: colors.dusk,
    maxWidth: 150,
  },
  txtDate: {
    fontSize: 12,
    color: colors.blueyGray,
    textAlign: 'right',
  },
});

type Props = {
  style?: ViewStyle;
  item: Chatroom;
  onPress?: () => void;
};

type State = {

};

class Shared extends Component<Props, State> {
  static defaultProps: Props = {
    style: styles.wrapper,
    item: {
      id: '',
      img: null,
      displayName: '',
      msg: '',
      count: '',
      date: null,
      status: false,
    },
  };

  render() {
    return (
      <View style={styles.container}>
        <TouchableOpacity
          activeOpacity={0.5}
          onPress={this.props.onPress}
        >
          <View style={this.props.style}>
            <View style={styles.imgWrapper}>
              {
                this.props.item.img
                  ? <Image style={styles.img} source={this.props.item.img}/>
                  : <Icon5 name="meh" size={40} color={colors.dusk} light/>
              }
              {
                this.props.item.status
                  ? <View style={styles.status}/>
                  : <View/>
              }
            </View>
            <View style={styles.viewContent}>
              <View style={styles.viewTop}>
                <Text style={styles.txtDisplayName}>{this.props.item.displayName}</Text>
                {
                  this.props.item.count
                    ? <View style={styles.viewCount}>
                      <Text style={styles.txtCount}>{this.props.item.count}</Text>
                    </View>
                    : <View/>
                }
              </View>
              <View style={styles.viewBottom}>
                <Text style={styles.txtMsg}>{this.props.item.msg}</Text>
                <Text style={styles.txtDate}>{moment(this.props.item.date).fromNow()}</Text>
              </View>
            </View>
          </View>
        </TouchableOpacity>
      </View>
    );
  }
}

export default Shared;
