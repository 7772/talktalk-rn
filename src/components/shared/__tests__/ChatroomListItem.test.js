import 'react-native';
import * as React from 'react';
import ChatroomListItem from '../ChatroomListItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';

describe('rendering test', () => {
  const wrapper = shallow(
    <ChatroomListItem />,
  );

  it('should render as expected', () => {
    expect(wrapper).toMatchSnapshot();
    // wrapper.setProps({ filled: false });
    // expect(wrapper).toMatchSnapshot();
  });
});
