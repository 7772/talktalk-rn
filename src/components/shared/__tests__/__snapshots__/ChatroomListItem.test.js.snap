// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test should render as expected 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.5}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 1,
          "borderColor": "rgb(247,248,251)",
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Object {
            "marginHorizontal": 20,
          }
        }
      >
        <FA5iconClass
          color="rgb(65,77,107)"
          light={true}
          name="meh"
          size={40}
          solid={false}
        />
        <View />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "paddingRight": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(65,77,107)",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(80,227,194)",
                "borderRadius": 8,
                "height": 16,
                "justifyContent": "center",
                "width": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 10,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 8,
            }
          }
        >
          <Text
            numberOfLines={2}
            style={
              Array [
                Object {
                  "color": "rgb(65,77,107)",
                  "fontSize": 12,
                  "maxWidth": 150,
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "rgb(134,154,183)",
                "fontSize": 12,
                "textAlign": "right",
              }
            }
          >
            49 years ago
          </Text>
        </View>
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
